test/jdk/javax/swing/JCheckBox/bug8032667.java

```diff
@@ -20,55 +20,72 @@
 import javax.swing.JComponent;
-import javax.swing.SwingUtilities;
+import javax.swing.JFrame;
 
 /* @test
  * @bug 8032667
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary [macosx] Components cannot be rendered in HiDPI to BufferedImage
- * @run applet/manual=yesno bug8032667.html
+ * @run main/manual bug8032667
  */
-public class bug8032667 extends JApplet {
+
+public class bug8032667 {
 
     static final int scale = 2;
     static final int width = 130;
     static final int height = 50;
     static final int scaledWidth = scale * width;
     static final int scaledHeight = scale * height;
 
-    @Override
-    public void init() {
-        SwingUtilities.invokeLater(new Runnable() {
+    private static final String INSTRUCTIONS = """
+            Verify that scaled components are rendered smoothly to image.
 
-            @Override
-            public void run() {
+            1. Run the test.
+            2. Check that Selected and Deselected JCheckBox icons are drawn smoothly.
+            If so, press PASS, else press FAIL.
+            """;
 
-                final Image image1 = getImage(getCheckBox("Deselected", false));
-                final Image image2 = getImage(getCheckBox("Selected", true));
-
-                Canvas canvas = new Canvas() {
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("bug8032667 Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(bug8032667::createAndShowGUI)
+                .build()
+                .awaitAndCheck();
+    }
 
-                    @Override
-                    public void paint(Graphics g) {
-                        super.paint(g);
-                        g.drawImage(image1, 0, 0, scaledWidth, scaledHeight, this);
-                        g.drawImage(image2, 0, scaledHeight + 5,
-                                scaledWidth, scaledHeight, this);
-                    }
-                };
+    public static JFrame createAndShowGUI() {
+        JFrame frame = new JFrame("bug8032667 HiDPI Component Test");
+        frame.setLayout(new BorderLayout());
+        frame.setBounds(0, 400, 400, 400);
 
-                getContentPane().add(canvas, BorderLayout.CENTER);
+        final Image image1 = getImage(getCheckBox("Deselected", false));
+        final Image image2 = getImage(getCheckBox("Selected", true));
+        Canvas canvas = new Canvas() {
+            @Override
+            public void paint(Graphics g) {
+                super.paint(g);
+                g.drawImage(image1, 0, 0, scaledWidth, scaledHeight, this);
+                g.drawImage(image2, 0, scaledHeight + 5,
+                        scaledWidth, scaledHeight, this);
             }
-        });
+        };
+        frame.getContentPane().add(canvas, BorderLayout.CENTER);
```

Comment:
It seems getContentPane is not needed and is obsoleted since jdk1.5...See this [comment](https://github.com/openjdk/jdk/pull/18390#discussion_r1533017080)